src/components/ReactTags.js

```diff
@@ -234,7 +236,7 @@ class ReactTags extends Component {
           ? suggestions[selectedIndex]
           : { id: query, [this.props.labelField]: query };
 
-      if (selectedQuery !== '') {
+      if (selectedQuery !== '' || this.props.allowEmpty === true) {
```

Comment:
why ` this.props.allowEmpty === true`, its anyways returning when `allowEmpty` is false